src/components/forms/EditGrid/EditGrid.stories.tsx

```diff
@@ -158,6 +160,49 @@ export const AddingItemInIsolationMode: Story = {
   },
 };
 
+const ItemBody: React.FC = () => {
+  const {values: formikValues} = useFormikContext();
+
+  const rawNamePrefix = useFieldConfig('');
+  if (!rawNamePrefix.endsWith('.')) throw new Error('Unexpected name prefix');
+  const namePrefix = rawNamePrefix.slice(0, -1);
+
+  const itemValues = getIn(formikValues, namePrefix);
+
+  return (
+    <p>
+      Formik data: <code>{JSON.stringify(itemValues)}</code>
+    </p>
+  );
+};
+
+export const NonEditableItemInIsolation: Story = {
+  args: {
+    enableIsolation: true,
+    getItemHeading: undefined,
+    getItemBody: () => <ItemBody />,
+    description: '',
+    emptyItem: undefined,
+    // The first item is uneditable, the second item is editable.
+    canEditItem: (_, index) => index === 1,
+    validate: () => Promise.resolve(),
```

Comment:
```suggestion
    validate: async () => {},
```
Use a consistent style of doing async/await - we want to hide the `Promise` concept as much as possible.